pkg/spdk/engine.go

```diff
@@ -227,6 +231,26 @@ func (e *Engine) Create(spdkClient *spdkclient.Client, replicaAddressMap map[str
 		}
 	}
 
+	// Validate the engien & replica sizes before creating the engine
+	for replicaName, replicaAddr := range replicaAddressMap {
+		replicaClient, err := GetServiceClient(replicaAddr)
+		if err != nil {
+			return nil, err
+		}
+		replica, err := replicaClient.ReplicaGet(replicaName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get replica %v from %v during engine creation", replicaName, replicaAddr)
+		}
+
+		if e.SpecSize != replica.SpecSize {
+			// not return error here
+			// it may cause infinite retry if the user doesn't fix the size issue
+			e.SpecSize = replica.SpecSize
+			e.log.Warnf("Engine spec size is not consistent with replica %v spec size, will use the replica spec size %v", replicaName, replica.SpecSize)
+			break
+		}
```

Comment:
Make sure all replicas have the same spec size first. And it should return an error if `e.SpecSize < replica.SpecSize`